File: web-interface/unittest/xivojson/mail.py
# -*- coding: utf-8 -*-
from __future__ import with_statement

__version__ = "$Revision$ $Date$"
__author__  = "Guillaume Bour <gbour@proformatique.com>"
__license__ = """
    Copyright (C) 2010  Proformatique

    This program is free software; you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation; either version 2 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License along
    with this program; if not, write to the Free Software Foundation, Inc.,
    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA..
"""
import unittest, pprint, cjson
from xivojson import *
# define global variables
from globals  import *


class Test40Mail(unittest.TestCase):
    def setUp(self):
        global IP, PORT, SSL, USERNAME, PASSWORD

        self.client = JSONClient(IP, PORT, SSL, USERNAME, PASSWORD)
        self.obj    = 'mail'

    def test_01_mail(self):
        (resp, data) = self.client.view(self.obj, 0)
        self.assertEqual(resp.status, 200)
        
        data = cjson.decode(data)
#        pprint.pprint(data)
        
        # EDIT
        with open('xivojson/mail.json') as f:
            content = cjson.decode(f.read())
            
#        print content
        (resp, data) = self.client.edit(self.obj, content)
        self.assertEqual(resp.status, 200)

File: web-interface/action/www/xivo/configuration/web_services/network/mail.php
<?php

#
# XiVO Web-Interface
# Copyright (C) 2010  Proformatique <technique@proformatique.com>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#
$access_category = 'network';
$access_subcategory = 'mail';

include(dwho_file::joinpath(dirname(__FILE__),'..','_common.php'));

$app = &$_XOBJ->get_application('mail');
$act = $_QRY->get('act');

switch($act)
{
	case 'edit':
	    $data = $app->_get_data_from_json();
	    
	    $canonical = "";
	    foreach($data['canonical'] as $rule)
	        $canonical .= $rule['pattern'] . ' ' . $rule['result'] . "\n";
	    $data['canonical'] = $canonical;
	    var_dump($data);
	    
		$status = $app->set($data) === true ? 200 : 400;

		$http_response->set_status_line($status);
		$http_response->send(true);
		break;

	default:
		$act = 'view';

		if(($view = $app->get()) === false)
		{
			$http_response->set_status_line(204);
			$http_response->send(true);
		}

        $canonical = array();
        foreach(preg_split('/\r?\n/', $view['canonical']) as $rule)
        {
            list($src, $dst) = preg_split('/\s+/', $rule);
            if(strlen($src) == 0)
                continue;
                
            $canonical[] = array('pattern' => $src, 'result' => $dst);
        }
        $view['canonical'] = $canonical;
        
		$_TPL->set_var('info',$view);
}

$_TPL->set_var('act',$act);
$_TPL->set_var('sum',$_QRY->get('sum'));
$_TPL->display('/xivo/configuration/generic');

?>
